src/display/annotation_layer.js

```diff
@@ -264,12 +264,40 @@ class AnnotationElement {
 
     container.style.left = `${(100 * (rect[0] - pageLLx)) / pageWidth}%`;
     container.style.top = `${(100 * (rect[1] - pageLLy)) / pageHeight}%`;
-    container.style.width = `${(100 * width) / pageWidth}%`;
-    container.style.height = `${(100 * height) / pageHeight}%`;
+
+    const { rotation } = data;
+    if (data.hasOwnCanvas || rotation === 0) {
+      container.style.width = `${(100 * width) / pageWidth}%`;
+      container.style.height = `${(100 * height) / pageHeight}%`;
+    } else {
+      this.setRotation(rotation, container);
+    }
 
     return container;
   }
 
+  setRotation(angle, container) {
+    container ||= this.container;
```

Comment:
Can't we just do the following?
```suggestion
  setRotation(angle, container = this.container) {
```